📝  overhaul of the documentation, now 4.5x bigger (better?)

baptistecolle opened the pull request:
# What does this PR do?

This is a complete overhaul of the documentation:
- We want from 1686 to 7565 words (4.5X bigger)
- We auto-generate documentation for our examples
    - This is auto generated from a .ipynb notebook using a custom scripts and the doc-builder converter feature https://moon-ci-docs.huggingface.co/docs/optimum-tpu/pr_144/en/howto/gemma_tuning
    - So when we add new notebook examples they will also be in the docs 😁
- New formatting and organization of the docs to make it easier to follow
- Added new tutorials, how-to, conceptual guides, and references following the diataxis method

What is missing (could be added):
- I think more examples would be nice, showing more diverse use cases
- I believe FAQ and glossary would be nice to add, but this PR is big enough already
- Guide and examples with Google Colab Pro as you can launch a v5e-1 TPU from there, so a one-click example would be nice
- An example using GCE VM on Colab via GCP Marketplace
- More diagrams and figures of the internal working of optimum-TPU to give some details would be interesting
- A how-to guide on adding new models for new contributors
- Docs for GKE is in the work and but not published yet as there are some blockers for that https://github.com/huggingface/optimum-tpu/blob/doc-deploy-gke/docs/source/howto/deploy-gke.md.
- The current preview docs for GKE are for CLI only. A GUI guide would be interesting too

More todos (to keep track of everything)
- Move current training tutorial to howto and make a training tutorial where all dependency are installed from scratch
- make sure us-docker.pkg.dev/deeplearning-platform-release/gcr.io/huggingface-pytorch-training-tpu.2.5.1.transformers.4.46.3.py310 is uploaded so it can be used. (also check for other GKE images)
- make the optimum-tpu ghcr.io latest point to tgi and not tgi-ie
- make the version of the container in the docs are dynamic so that the docs is always pointing to the latest version
- inject duplicate content dynamically (to make maintenance easier)
- assert max-batch-prefill-tokens ≤ max-input-length * max_batch_size
- refactor the name of the files (.mdx) so they match their title better
- look at kaggle as they also support TPU so adding a example with Kaggle would be nice
- add links from other huggingface project to optimum-tpu
    - https://huggingface.co/docs/transformers/perf_train_tpu_tf
    - https://huggingface.co/docs/accelerate/en/concept_guides/training_tpu
    - https://huggingface.co/docs/accelerate/en/basic_tutorials/tpu
    - Look at other project such as TRL, peft if we can add a page or a link to optimum-tpu there
 - Add example on diffusers with TPU
 - Add to CI `markdown-link-check docs/source/**/*.mdx` 

### New Files Added
- `docs/scripts/auto-generate-examples.py`
- `docs/scripts/examples_list.yml`
- `docs/source/conceptual_guides/difference_between_jetstream_and_xla.mdx`
- `docs/source/conceptual_guides/tpu_hardware_support.mdx`
- `docs/source/contributing.mdx`
- `docs/source/howto/advanced-tgi-serving.mdx` 
- `docs/source/howto/deploy_instance_on_ie.mdx`
- `docs/source/howto/installation_inside_a_container.mdx`
- `docs/source/installation.mdx`
- `docs/source/optimum_container.mdx`
- `docs/source/reference/fsdp_v2.mdx`
- `docs/source/reference/tgi_advanced_options.mdx`
- `docs/source/tutorials/inference_on_tpu.mdx`
- `docs/source/tutorials/tpu_setup.mdx`
- `docs/source/tutorials/training_on_tpu.mdx`

### Modified Files
- `docs/source/howto/training.mdx`
- `docs/source/index.mdx`
- `docs/source/supported-architectures.mdx`

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/optimum-tpu/pr_144). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

baptistecolle commented:
BTW just for reference. We also now link to the optimum-tpu docs from:
- TGI docs: PR review in progress: https://github.com/huggingface/text-generation-inference/pull/2907
- HF Google Cloud docs: https://huggingface.co/docs/google-cloud/en/tpu (already live)
The goal is to increase visibility of the doc

tengomucho commented on docs/source/tutorials/tpu_setup.mdx:
```diff
@@ -0,0 +1,76 @@
+
+1. Select TPU type:
+   - We'll use a TPU `v5e-8` (corresponds to a v5litepod8). This is a TPU node containing 8 v5e TPU chips
+   - For detailed specifications about TPU types, refer to our TPU types documentation
```
provide a link

tengomucho commented on docs/source/tutorials/tpu_setup.mdx:
```diff
@@ -0,0 +1,76 @@
+   - Learn how to start training ML models on TPU
+
+Choose the tutorial that best matches your immediate needs:
+- For deploying existing models, start with Model Serving
+- For training new models, begin with Model Training
```
provide links

tengomucho commented on docs/source/tutorials/training_on_tpu.mdx:
```diff
@@ -0,0 +1,87 @@
+
+## 1. Start the Jupyter Container
+
+Launch the container with the following command:
```
you need to clone the optimum tpu git, install the jupyter notebook and then you can run it, but you will need to mount the notebook too.

baptistecolle replied:
I dont get it, WDYM?. We built us-docker.pkg.dev/deeplearning-platform-release/gcr.io/huggingface-pytorch-training-tpu.2.5.1.transformers.4.46.3.py310 and this already has jupyter notebook installed and the example is embeded there? So user does not have to do more. The commands start the jyupter webui so user can use directly in seconds

baptistecolle replied:
Actually, I thought more about and I get what you are saying and you are right. Indeed, it does not make sense to have the tutorial directly use the container. A better tutorial  would manually install optimum tpu and all other dependency and then show code steps by steps on how to do training like the fine-tune examples. Using the container and the notebook directly should be moved to howto i think and be more like "here is a container that has everything package for you for training and some examples included". So the main tutorial would be setting up training from scratch on a TPU vm. I edited the PR description and added this as a todo for the docs for a next PR

tengomucho commented on docs/source/tutorials/training_on_tpu.mdx:
```diff
@@ -0,0 +1,87 @@
+docker run --rm --net host --privileged \
+    -v$(pwd)/artifacts:/tmp/output \
+    -e HF_TOKEN=${HF_TOKEN} \
+    us-docker.pkg.dev/deeplearning-platform-release/gcr.io/huggingface-pytorch-training-tpu.2.5.1.transformers.4.46.3.py310 \
```
I do not think we should provide a link to an image url that does not exist yet 😢

baptistecolle replied:
Let us work asap to get that ready but i think merging docs already is fine. I can adjust post PR and provide the url to the HF artificat repisotry as we are not live on the us-docker.pkg.dev/deeplearning-platform-release. Is that fine by you?

tengomucho commented on docs/source/tutorials/inference_on_tpu.mdx:
```diff
@@ -0,0 +1,153 @@
+- `--privileged`: Required for TPU access
+- `--net host`: Uses host network mode
+- `-v ~/hf_data:/data`: Volume mount for model storage
+- `-e SKIP_WARMUP=1`: Disables warmup for quick testing (not recommended for production)
```
remove this

baptistecolle replied:
I prefer keeping it but yes let me work on that after this pr

> I see your point. Let me try to find a way to inject that into each page to prevent duplication and simplify maintainability. Do you mind if i do that in another PR. I did not see that option in the docs of [HF doc-builder](https://github.com/huggingface/doc-builder) so i wanted to ask the team

tengomucho reviewed (commented):
Thanks for the huge work! Some general comments
- I would prefer to avoid repetition: having information repeated in several places can be confusing and it is harder to maintain. E.g.: docker arguments, TGI args
- you specify version numbers, I think it would be best if we could generate that, otherwise it will be a burden to maintain
- Try to keep titles and toc tree in sync
- There is a bit of repetition between the tutorials and howtos. Maybe you can rationalize that.
- the conceptual guides should be more focused on optimum-tpu IMO, what do you think?

baptistecolle commented on docs/source/conceptual_guides/difference_between_jetstream_and_xla.mdx:
```diff
@@ -0,0 +1,17 @@
+# Differences between JetStream and PyTorch XLA
```
the conceptual guides should be more focused on optimum-tpu IMO, what do you think?
- I don't think so. For example in transformers they explain the different style of attention https://huggingface.co/docs/transformers/en/attention. For me one of the goal of conceptual guides is to explain different concepts that are needed for the end-user to be able to use and understand our library optimum-tpu. 

That's why i prefer to keep it. I modified the page quite a bit let me know if this still does not make sense for you

baptistecolle commented on docs/source/conceptual_guides/tpu_hardware_support.mdx:
```diff
@@ -0,0 +1,54 @@
+# TPU hardware support
+Optimum-TPU support and is optimized for V5e, V5p, and V6e TPUs.
```
FYI, i removed it here i however left the information about v5p elsewhere in the page like how much memory v5p has, i hope this is fine by you as this information might be useful to a end-user

baptistecolle commented on docs/source/conceptual_guides/tpu_hardware_support.mdx:
```diff
@@ -0,0 +1,54 @@
+For example, a v5litepod-8 is a v5e TPU with 8 tpus.
+
+## Memory on TPU
+The HBM (High Bandwidth Memory) capacity per chip is 16gb for V5e, V5p and 32gb for V6e. So a v5e-8 (v5litepod-8), has 16gb*8=128gb of HBM memory
```
I actually always do this mistake so thanks for pointing it out 😅

tengomucho replied:
V5e -> v5e V5p -> V5e

baptistecolle commented on docs/source/conceptual_guides/tpu_hardware_support.mdx:
```diff
@@ -0,0 +1,54 @@
+## Memory on TPU
+The HBM (High Bandwidth Memory) capacity per chip is 16gb for V5e, V5p and 32gb for V6e. So a v5e-8 (v5litepod-8), has 16gb*8=128gb of HBM memory
+
+## Performance on TPU
```
as it in conceptual guide, i believe keeping it is better as it give a way for user to know what are the difference between all the tpu hardware and what are the performance implication

baptistecolle commented on docs/source/conceptual_guides/tpu_hardware_support.mdx:
```diff
@@ -0,0 +1,54 @@
+# Next steps
+For more information on the different TPU hardware, you can look at:
+https://cloud.google.com/tpu/docs/v6e
+https://cloud.google.com/tpu/docs/v5p
```
I prefer to keep it here for reference for the users as the information is universal

baptistecolle commented on docs/source/conceptual_guides/tpu_hardware_support.mdx:
```diff
@@ -0,0 +1,54 @@
+During the TPU VM creation use the following TPU VM base images for optimum-tpu:
+- v2-alpha-tpuv6e (TPU v6e) (recommended)
+- v2-alpha-tpuv5 (TPU v5p) (recommended)
+- v2-alpha-tpuv5-lite (TPU v5e) (recommended)
```
I am not sure what you mean here? I have in the next paragraph a link back to the setup tpu tutorial so that users can see those TPU base Vm image in action, so i think this answer you question right?

tengomucho replied:
e.g.  ghcr.io/huggingface/optimum-tpu

baptistecolle replied:
This page is more about the TPU hardware and not directly about our containers, I mention ghcr.io/huggingface/optimum-tpu elsewhere in the docs. This is about base TPU vm image (so the OS). Maybe title of the section is not great as you were confused. DO you have suggestion?

baptistecolle commented on docs/source/howto/advanced-tgi-serving.mdx:
````diff
@@ -0,0 +1,71 @@
+If you encounter `Backend(NotEnoughMemory(2048))`. 
+Here are some solutions that could help with reducing memory usage in TGI:
+
+bash
+```
+docker run -p 8080:80 \
+        --shm-size 16G \
+        --privileged \
+        --net host \
+        -e QUANTIZATION=1 \
+        -e MAX_BATCH_SIZE=2 \
+        -e LOG_LEVEL=text_generation_router=debug \
+        -v ~/hf_data:/data \
+        -e HF_TOKEN=$(cat ~/.cache/huggingface/token) \
+        -e SKIP_WARMUP=1 \
+        ghcr.io/huggingface/optimum-tpu:v0.2.3-tgi \
+        --model-id google/gemma-2b-it \
+        --max-input-length 512 \
+        --max-total-tokens 1024 \
+        --max-batch-prefill-tokens 512
+        --max-batch-total-tokens 1024
+```
````
I prefer to keep duplicate information because of the Proximity Principle - keeping related information close together where it will be used, even if that means some duplication.

tengomucho replied:
Yes but the sentence does not make much sense to me. I am reading "How to solve memory requirements: command line with loads of args followed by description of args". I think you have to say that if you see the mentioned error, then you can try to reduce some of the parameters that will consequently consume less HBM TPU memory. In particular you can reduce batch size, input length and total tokens, etc etc, and then maybe you show an example.

baptistecolle replied:
Ah ok i get it! Let me if the new version makes more sense. An resolve conversation if it does

tengomucho commented on docs/source/howto/advanced-tgi-serving.mdx:
```diff
@@ -0,0 +1,71 @@
+
+## How to solve memory requirements
+
+If you encounter `Backend(NotEnoughMemory(2048))`. 
+Here are some solutions that could help with reducing memory usage in TGI:
```
BTW you need to rephrase this

baptistecolle commented on docs/source/howto/advanced-tgi-serving.mdx:
```diff
@@ -0,0 +1,71 @@
+
+To reduce memory usage, you want to try a smaller number for  `--max-input-length`, `--max-total-tokens`, `--max-batch-prefill-tokens`, and `--max-batch-total-tokens`. 
+
+<Tip warning={true}>
+`max-batch-prefill-tokens ≤ max-input-length * max_batch_size`. Otherwise, you will have an error as the configuration does not make sense. If the max-batch-prefill-tokens were bigger, then you would not be able to process any request
+</Tip>
```
Indeed let's leave this here for now but yeah we should make a small PR for that that would be nicer

baptistecolle commented on docs/source/howto/deploy_instance_on_ie.mdx:
```diff
@@ -0,0 +1,60 @@
+
+![Configure endpoint](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/optimum/tpu/ie_config_new_endpoint.png)
+
+Note: We automatically select the optimal hardware and configuration for each model. For google/gemma-2b-it, being a smaller model, we select a 1-chip TPU (TPU v5e-1) since 16GB of HBM (High Bandwidth Memory) is sufficient to serve the 2B model. This ensures cost-efficient resource allocation without unnecessary computing expenses.
```
I prefer describing our choices i believe it is always nice for users to understand how things work under the hood especially when they need to also to deploy and maintain it

baptistecolle commented on docs/source/howto/gcloud_cli.mdx:
````diff
@@ -58,25 +57,3 @@ gcloud compute tpus tpu-vm ssh optimum-tpu-get-started --zone=us-west4-a
 $ >
 ```
 
-## Setting up the instance to run AI workloads on TPUs
````
I moved it somewhere else. but actually yeah let me add it back here (so that would duplicate content) but it would be nice to say that you have your tpu instance you can install optimum-tpu if you wish

baptistecolle commented on docs/source/howto/serving.mdx:
```diff
@@ -1,24 +1,60 @@
 # Deploying a Text-Generation Inference server on a Google Cloud TPU instance
 
-## Context
+Serving Large Language Models (LLMs) on Optimum-TPU is mainly done via TGI. TGI is a highly optimized serving engine that enables LLMs to be served in the most performant way. At Optimum-TPU, we have optimzied TGI for the TPU hardware. TGI also offers an openAI-compatible API, making it easy to integrate with numerous tools.
```
Do you like the new version better?

baptistecolle commented on docs/source/howto/training.mdx:
```diff
@@ -19,12 +19,15 @@ pip install optimum-tpu -f https://storage.googleapis.com/libtpu-releases/index.
+   - Check our [LLaMA fine-tuning notebook](https://github.com/huggingface/optimum-tpu/blob/main/examples/language-modeling/llama_tuning.ipynb) for detailed instructions
+
+## Tips
+We recommend fine-tuning using bf16 on TPU, as those operations are often extremely fast while keeping the training stable and precise enough.
```
It is in our jyupter notebooks. this was more of a general advice but i can keep those advice for the notebook only do you prefer that or should i leave it here?

baptistecolle commented on docs/source/index.mdx:
```diff
@@ -16,33 +16,22 @@ limitations under the License.
+</Tip>
 
-The API provides the overall same user-experience as Hugging Face transformers with the minimum amount of changes required to target performance for inference and training.
+Optimum TPU provides all the necessary machinery to leverage and optimize AI workloads running on [Google Cloud TPU devices](https://cloud.google.com/tpu/docs). Optimum-TPU is a Huggingface solution to optimize Huggingface products for the TPU platform. This allows users to use Huggingface features and easy-to-use libraries on TPU with the best performance. We currently optimize transformers and  ...
```
😭 Yes indeed that would be better and the proper way to do it. Btw i checked transformers by curiosity and it is not consistent there. The also Hugging Face and HuggingFace. I put HuggingFace with proper capatilization everywhere if that's fine by you

baptistecolle commented on docs/source/installation.mdx:
````diff
@@ -0,0 +1,22 @@
+
+```bash
+$ python3 -m pip install optimum-tpu -f https://storage.googleapis.com/libtpu-releases/index.html
+$ export PJRT_DEVICE=TPU
````
You mean the export PJRT_DEVICE=TPU?

baptistecolle commented on docs/source/reference/fsdp_v2.mdx:
```diff
@@ -0,0 +1,103 @@
+
+### Integration with Trainer
+
+FSDP_v2 configuration can be directly used with the TPUTrainer:
```
Indeed, thanks for spotting it 😅

baptistecolle commented on docs/source/reference/fsdp_v2.mdx:
```diff
@@ -0,0 +1,103 @@
+
+## Troubleshooting
+
+Common issues and solutions:
```
I added a link to our notebooks for best practice on how to do fine-tuning with optimum-tpu

baptistecolle commented on docs/source/tutorials/inference_on_tpu.mdx:
````diff
@@ -0,0 +1,153 @@
+### SSH Access
+First, connect to your TPU instance via SSH.
+
+### Install Required Tools
+
+Install the Hugging Face Hub CLI:
+```bash
+pip install huggingface_hub
+```
+
+### Authentication
+
+Log in to Hugging Face:
+```bash
+huggingface-cli login
+```
````
You are right! and i better understand your previous comments 
> either use <your_hf_token_here> or modify it further down in the page.

Ok so the idea in the tutorial is to offer one-way to do things that we recommend. This is my prefer way for the users. But in the howto section let me add. In the howto we shoudl be as exhaustive as possible and say that there a multiple way to setup the HF_TOKEN. 

I hope this answer your question.

baptistecolle commented on docs/source/howto/serving.mdx:
```diff
@@ -1,24 +1,60 @@
+        --net host \
+        -e LOG_LEVEL=text_generation_router=debug \
+        -v ~/hf_data:/data \
+        -e HF_TOKEN=$(cat ~/.cache/huggingface/token) \
```
Also look at my comment below because you talk to that after. 

> You are right! and i better understand your previous comments
> 
> > either use <your_hf_token_here> or modify it further down in the page.
> 
> Ok so the idea in the tutorial is to offer one-way to do things that we recommend. This is my prefer way for the users. But in the howto section let me add. In the howto we shoudl be as exhaustive as possible and say that there a multiple way to setup the HF_TOKEN.
> 
> I hope this answer your question.

baptistecolle commented on docs/source/tutorials/inference_on_tpu.mdx:
```diff
@@ -0,0 +1,153 @@
+
+### Understanding the Configuration
+
+Key parameters explained:
```
I see your point. Let me try to find a way to inject that into each page to prevent duplication and simplify maintainability. Do you mind if i dont that in another PR. I did not see that option in the docs of [HF doc-builder](https://github.com/huggingface/doc-builder) so i wanted to ask the team

baptistecolle commented on docs/source/tutorials/inference_on_tpu.mdx:
```diff
@@ -0,0 +1,153 @@
+
+Note on warmup:
+- TGI performs warmup to compile TPU operations for optimal performance
+- For this tutorial, we use `SKIP_WARMUP=1` to experiment quickly with TPU, but this means the first request will be slower as compilation happens on demand
+- For production use, remove the `SKIP_WARMUP=1` flag to improve performance
```
I like it in the tutorial because the goal is for the user to see results as fast as possible and warmups takes forever

I however removed it everywhere else except for the reference page and added a <Tip warning={true}> in the tutorial about it

baptistecolle commented on docs/source/tutorials/inference_on_tpu.mdx:
```diff
@@ -0,0 +1,153 @@
+
+Your TGI server is now ready to serve request.
+
+### Local Testing
```
I dont get it it is local because we are call 0.0.0.0 and not a remote ip?

baptistecolle commented on docs/source/tutorials/inference_on_tpu.mdx:
```diff
@@ -0,0 +1,153 @@
+
+To query from outside the TPU instance:
+
+![External IP TPU](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/optimum/tpu/get_external_ip_tpu.png)
```
added that to the gcloud cli guide thanks

baptistecolle commented on docs/source/tutorials/tpu_setup.mdx:
```diff
@@ -0,0 +1,76 @@
+## Step 1: Enable TPU Access
+
+1. Navigate to the TPU dashboard at: https://console.cloud.google.com/compute/tpus
+   - Note: You'll need to enable the TPU API if you haven't already
```
I changed that everywhere thx

baptistecolle commented:
Thanks a lot @tengomucho for all the feedbacks took me a while to integrate everything 😅 and i still believe there are some stuff that I am not sure what you mean exactly.

As there are a lot of comments already. Please mark them as resolve if you believe they do not require further action from my part so I can keep better track of everything left to.

I believe I implemented most of the changes and this should be ready for another review.

> Thanks for the huge work! Some general comments
> 1) I would prefer to avoid repetition: having information repeated in several places can be confusing and it is harder to maintain. E.g.: docker arguments, TGI args
> 2) you specify version numbers, I think it would be best if we could generate that, otherwise it will be a burden to maintain
> 3) Try to keep titles and toc tree in sync
> 4) There is a bit of repetition between the tutorials and howtos. Maybe you can rationalize that.
> 5) the conceptual guides should be more focused on optimum-tpu IMO, what do you think?

1) I would prefer to keep information close to where it is needed. So that people don't have to jump around in the docs to find everything, that's why there are some duplication
> I see your point. Let me try to find a way to inject that into each page to prevent duplication and simplify maintainability. Do you mind if i do that in another PR. I did not see that option in the docs of [HF doc-builder](https://github.com/huggingface/doc-builder) so i wanted to ask the team
> I prefer to keep duplicate information because I believe keeping related information close together where it will be used is important, even if that means some duplication. This offer a better user-experience and user may come to the documentation from any single page so they order it is read is not linear

2) Yes, do you mind if i do that in another PR where i look also add 1) and how to inject everything dynamically

3) Great feedback, I modified that and they are now sync
Question for you. I would like to modify a lot of the files name in the docs/source folder (refactoring) because the name of the file are not the most descriptive and structured. I believe it is best done in another PR or do you think doing it here is best as like a last commit after the approval is best. If i do it here i will mess all the git diff so more difficult to review it no?

4) Sure
first of all i recommend this talk on docs from one of the maintainer of django (Daniele Procida)
https://www.youtube.com/watch?v=t4vKPhjcMZg also look up other ressources tied to Daniele Procida 
He talks about https://diataxis.fr/ a way to structure docs
Let me however answers your question directly. Why the overlap between tutorials and howto. Tutorial offer a way for the user to accomplish a task (we dont talk tradeoff the different options and so on, only a recipe to follow). Howto, we explain the rational between each action why do we do this steps (keeping with the analogy of cooking, why do we bake a cake at a given temperature for example). They have overlap but how to goes a step above just the tutorial and how serving for example actually work on optimum tpu not just only how to achieve it. About optimum-tpu i dont know if the current pages are perfect on that regards but the goal for them would be to encapsulate that philosophy. I hope this answers your question

5) it links to 4) too
I answered that in a comment above
> the conceptual guides should be more focused on optimum-tpu IMO, what do you think?
> 
> I don't think so. For example in transformers they explain the different style of attention https://huggingface.co/docs/transformers/en/attention. For me one of the goal of conceptual guides is to explain different concepts that are needed for the end-user to be able to use and understand our library optimum-tpu.

baptistecolle commented:
I also added a "More todos (to keep track of everything)" section to the original PR description about the next tasks to do related to this PR, to prevent having everything in this PR and keep it from being more monstrous 👹

tengomucho commented on docs/source/howto/advanced-tgi-serving.mdx:
```diff
@@ -52,19 +60,26 @@ To reduce memory usage, you want to try a smaller number for  `--max-input-lengt
 - `--net host`: Uses host network mode
-- `-v ~/hf_data:/data`: Volume mount for model storage
-- `-e SKIP_WARMUP=1`: Disables warmup for quick testing (not recommended for production)
+Those are needed to run a TPU container so that the container can properly access the TPU hardware
```
full stop 😆

tengomucho commented on docs/source/tutorials/training_on_tpu.mdx:
```diff
@@ -16,27 +16,40 @@ The `huggingface-pytorch-training-tpu` Docker image provides a pre-configured en
+- `--shm-size 16GB`: Increase default shared memory allocation
 - `--net host`: Use host network mode for optimal performance
 - `--privileged`: Required for TPU access
+Those are needed to run a TPU container so that the container can properly access the TPU hardware
```
full stop?

baptistecolle commented:
@tengomucho I think i have address all the comments now! 

1) I removed those parts that were indeed not actionnable.
```
## Tips
We recommend fine-tuning using bf16 on TPU, as those operations are often extremely fast while keeping the training stable and precise enough.
```
```
## Troubleshooting

Common issues and solutions:

1. Out of Memory (OOM):
   - Enable gradient checkpointing
   - Reduce batch size
   - Use a smaller sequence length

2. Training Speed:
   - Ensure proper batch size optimization
   - Monitor TPU device utilization
   - Check for communication bottlenecks
 ```
2) I remove the page about the training container that is not live yet
3) I redid the training page. It is a full contain example without docker on how to do training on TPU with optimum so please look at https://moon-ci-docs.huggingface.co/docs/optimum-tpu/pr_144/en/tutorials/training_on_tpu

tengomucho commented:
I like it a lot more! Please fix the broken links on the main page as discussed and then this LGTM!

tengomucho commented on docs/source/conceptual_guides/tpu_hardware_support.mdx:
```diff
@@ -11,19 +11,7 @@ TPU version:
-- Peak compute per Pod: Total computing power when multiple chips work together. These indicate performance at scale for large training or serving jobs.
-
-The actual performance you achieve will depend on your specific workload characteristics and how well it matches these hardware capabilities.
+The HBM (High Bandwidth Memory) capacity per chip is 16GB for v5e, v5p and 32GB for v6e. So a v5e-8 (v5litepod-8), has 16GB*8=128GB of HBM memory
```
full st... oh, whatever

tengomucho commented on docs/source/howto/advanced-tgi-serving.mdx:
```diff
@@ -41,13 +41,13 @@ More information on tensor parralelsim can be found here https://huggingface.co/
-- `--privileged`: Required for TPU access
-- `--net host`: Uses host network mode
-Those are needed to run a TPU container so that the container can properly access the TPU hardware
+- `--shm-size 16GB`: Increase default shared memory allocation.
```
❤️

tengomucho reviewed (commented):
note there are some failure in dock build

baptistecolle commented:
There was a typo that cause the build to fail. This is fixed now. Also all the links between pages have been checked to prevent redirecting to wrong pages (added a todo to here to make this part of the CI). I think now everything should be okay. Let me know if it is missing anything else.

tengomucho reviewed (approved):
LGTM

baptistecolle commented:
@pagezyhf Let me know if you see any changes or potential improvements with the docs. I can also do separate PRs if you think some pages are missing and need to be added
